app/services/analytics_events.rb

```diff
@@ -565,6 +565,16 @@ def idv_verify_in_person_troubleshooting_option_clicked(flow_path:, **extra)
     )
   end
 
+  # @param [Hash] variant The IPP CTA Variant that was shown to the user
+  # The user was shown a variant of the IPP CTA
+  def idv_ipp_cta_variant(variant:, **extra)
+    preserve_testing_value({ ipp_cta_variant: variant })
```

Comment:
I am *very* skeptical of putting this side effect here. Again, the only purpose of this file is to provide some static definitons we can generate documentation from, I think it would be clearer to move this logic *up* to the call sites

Take a look at #7685 as an example of removing event-specific side effects.

Reply:
I was struggling with the best way to persist a value in session from the doc auth React flow. Could I put some time on your calendar to discuss?

Reply:
Yup! Go for it

Reply:
Could we assign the A/B test bucket to the session before the page is loaded. And then if we need to add that to events for in-person proofing, we could use the existing [`Idv::AnalyticsEventsEnhancer`](https://github.com/18F/identity-idp/blob/main/app/services/idv/analytics_events_enhancer.rb) class to add the variant in dynamically to every event for the in-person flow, since it was designed to add common details to all IPP events.